packages/circuit-ui/components/IconButton/IconButton.tsx

```diff
@@ -40,7 +40,7 @@ const sizeStyles = (size: IconButtonProps['size'] = 'giga') => (
 ): SerializedStyles => {
   const sizeMap = {
     kilo: theme.spacings.byte,
-    giga: theme.spacings.mega,
+    giga: theme.spacings.kilo,
```

Comment:
Could we automatically set the icon size based on the button size on line 58?

Reply:
Do you mean pass the relevant `size` down to the children (a single icon) so that it doesn't need to be added manually?

Can we do this while avoiding this? 👇 

```tsx
<IconButton>
  {({ size }) => <Icon size={size} />}
</IconButton>
```

Reply:
Yes. The IconButton already clones its child to add the `role: 'presentation'` prop, you can add the size there as well.

Reply:
Nice, I missed that! Done in 35969d1.

The only thing I'm wondering is, what if the icon doesn't exist in the relevant size? (e.g. an IconButton with a "AddProduct" which only exists in 24px) Would TS throw an error? No right? So we'd just have an icon with the wrong size, and probably could spot it in development

Reply:
That's true, TypeScript wouldn't catch that 😕

Reply:
☝️ actually it would be filtered out just like for the other issue we had with the `size` prop in the DOM, I guess